### PR TITLE
Add libraries to message threads that have them

### PR DESCRIPTION
**Pull request by ryanpbrewster**
@andrewconner

### CODE REVIEW

File: kifi-backend/modules/eliza/app/com/keepit/eliza/controllers/shared/SharedWsMessagingController.scala [hunk 1]
```diff
@@ -76,18 +78,22 @@ class SharedWsMessagingController @Inject() (
...
-          SafeFuture(socket.channel.push(Json.arr(
+          libOpt <- keepOpt.flatMap(k => Library.decodePublicId(k.libraryId).toOption).map { libId =>
+            shoebox.getBasicLibraryDetails(Set(libId), idealImageSize = ProcessedImageSize.Small.idealSize, viewerId = Some(socket.userId)).map(res => res.values.headOption)
+          }.getOrElse(Future.successful(None))
+        } yield {
+          socket.channel.push(Json.arr(
             "thread", Json.obj(
               "id" -> threadId,
               "uri" -> thread.url,
               "keep" -> keepOpt,
```
Comment (andrewconner) [inline kifi-backend/modules/eliza/app/com/keepit/eliza/controllers/shared/SharedWsMessagingController.scala:92]: Any chance we could do a stripped down `KeepInfo`? It's at least in a format clients are used to, so they'll be able to reuse parsing. For instance, on mobile, they parse json into static models.

Comment (andrewconner) [inline kifi-backend/modules/eliza/app/com/keepit/eliza/controllers/shared/SharedWsMessagingController.scala:92]: More importantly, what to include will be design-centric. I could imagine something similar to a keep card up top, and so sticking to something _like_ `KeepInfo` would probably get close.

That said, I'd hold off until there's a design — or, send a keep info and trim down later (as to not waste your time while we wait).

Comment (leogrim) [inline kifi-backend/modules/eliza/app/com/keepit/eliza/controllers/shared/SharedWsMessagingController.scala:92]: everything Andrew said

**Comment by andrewconner** [comment]
Follow up, expect specifics today about what data needs to come down.

**Comment by andrewconner** [comment]
... no design yet :(. Am still trying to get one so I can give you what's needed on clients. Renew your "today" expectations.

**Comment by andrewconner** [comment]
Send down an object containing at least these things:
- Keep title
- Keep id
- Keep url
- Keep image URL
- Keep note
- Keep hashtags (seq of strings)
- Original keeper (user)
- Keep date
- Library title
- Library path/url
- Library organization
- Library color
- Library visibility
- Library id
- Whether current user can edit/unkeep/view library

Keep in mind that it may be possible for a user to not be in the library, but be in the discussion. In this case, they can see the existence of the library, but will not be able to click to see it.

You can probably reuse existing classes, but these are the things that will be used in the design.

@ryanpbrewster

**Pull request closed without merging**